core/src/main/java/io/fabric8/maven/core/service/openshift/OpenshiftBuildService.java

```diff
@@ -245,8 +245,8 @@ private String updateBuildConfig(OpenShiftClient client, String buildName, Build
         if (!Objects.equals(buildStrategy, spec.getStrategy()) || !Objects.equals(buildOutput, spec.getOutput())) {
             client.buildConfigs().withName(buildName).edit()
                     .editSpec()
-                    .withStrategy(buildStrategy)
-                    .withOutput(buildOutput)
+                    .withStrategy(spec.getStrategy())
+                    .withOutput(spec.getOutput())
```

Comment:
I'm really not sure why this should fix something, as this is a no-op after all: You are re-setting the same strategy and output as it was there already. The whole idea of this method is to _change_ the strategy (docker or binary s2i build) and the output stream if changed.

@zonArt could you please elaborate what are you trying to achieve ?

Reply:
@rhuss : I read the related issue file by @zonArt , I think he wants to have his pushSecrets kept in `BuildConfig` to push/pull from some private registry, maybe that's why he is setting same strategy and output itself.

Agreed that this doesn't align with plugin's flow at all. But is there some way to configure plugin to include these pushSecrets into BuildConfig?

Reply:
@rohanKanojia You got it totally right, being able to keep the configuration of the buildconfig was the needed behavior thus this PR which was maybe not intended to be merged as is but more a way to request support on how to customize those objects or to find a way to keep provided ones (maybe with a flag)